packages/eslint-plugin/docs/rules/no-unnecessary-type-parameters.mdx

```diff
@@ -87,6 +87,108 @@ This is because the type parameter `T` relates multiple methods in the `T[]` tog
 Therefore, this rule won't report on type parameters used as a type argument.
 That includes type arguments given to global types such as `Array` (including the `T[]` shorthand and in tuples), `Map`, and `Set`.
 
+## FAQ
+
+### I'm using the type parameter inside the function, so why is the rule reporting?
+
+You might be surprised to that the rule reports on a function like this:
+
+```ts
+function f<T extends string>(string1: T): void {
```

Comment:
I would love if someone has a better example for this. I didn't just copy the example from https://github.com/typescript-eslint/typescript-eslint/issues/9828 since it didn't seem very organic, but I'm not sure this is any better 🤔